thriftpy/protocol/compact.py

```diff
@@ -58,7 +58,10 @@ def write_varint(trans, n):
         else:
             out.append((n & 0xff) | 0x80)
             n = n >> 7
-    data = array.array('B', out).tostring()
+
+    # 3.2+ uses tobytes(). Thriftpy doesn't support Python 3 pre-3.3
+    a = array.array('B', out)
+    data = a.tobytes() if PY3 else a.tostring()
```

Comment:
I'd recommend doing the check once at the top of the file instead of once per call:
```
if PY3:
  tobytes = array.tobytes
else:
  tobytes = array.tostring

...
  data = tobytes(a)
...
```
That said, I don't have merge writes - so take suggestions with a grain of salt  =u)